### ISSUE TITLE
sparse tensors

### ISSUE THREAD

**Issue by guidocalvano** [tensorflow/tfjs#906]
Are sparse tensors under development? 

They would make my life so much easier...

To get help from the community, check out our [Google group](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs).

**Comment by rthadur** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!

**Comment by juhoojala** [comment]
Thanks for this great library!

I'm also interested in this. For me it would be useful for a non-ML-related optimization problem.  

The python version has it, and I believe that javascript users would benefit similarily of the functionality as python users do
https://www.tensorflow.org/api_docs/python/tf/sparse/SparseTensor

**Comment by TheNewSound** [comment]
I also have the same feature request.

Please support creation of sparse tensors and operations like '`tensorflow::ops::SparseMatMul`'
Link to C++ Tensorflow: [https://www.tensorflow.org/api_docs/cc/class/tensorflow/ops/sparse-mat-mul](https://www.tensorflow.org/api_docs/cc/class/tensorflow/ops/sparse-mat-mul)

**Comment by okpatil4u** [comment]
@tafsiri I am using tfjs for graph neural networks in browsers. Sparse tensors could help speed it up considerably.

**Comment by dallman2** [comment]
I would also like to be able to use sparse tensors. Currently, I am boxed into padding my data with junk during preprocessing to avoid having a ragged tensor. 

The use case is very similar to sentiment analysis, with words being mapped to sequential states of a football game and the sentiment being mapped to the winner of the game. Maybe I am not thinking about this correctly, but it seems that manipulating a sparse tensor during runtime would be much easier than a ragged one.
